src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -48,12 +49,60 @@ export class RuntimeLanguageContext {
     }
 
     public convertLanguage(languageId?: string) {
-        languageId = languageId === CodeWhispererConstants.typescript ? CodeWhispererConstants.javascript : languageId
-        if (!languageId) {
+        let mappedId: string | undefined
+        switch (languageId) {
+            case CodeWhispererConstants.typescript:
+                mappedId = CodeWhispererConstants.javascript
+                break
+            case 'javascriptreact':
```

Comment:
You can replace string 'javascriptreact' with a variable name

Reply:
yea I will

Reply:
Is it possible to simplify the language id conversion logic?

Reply:
we can remove the undefined case
```
public convertLanguage(languageId: string): string {
        let mappedId: string
        switch (languageId) {
            case CodeWhispererConstants.typescript:
                mappedId = CodeWhispererConstants.javascript
                break
            case 'javascriptreact':
                mappedId = CodeWhispererConstants.jsx
                break
            default:
                mappedId = languageId
                break
        }

        return mappedId
    }
```

Reply:
other workaround is to have a map storing these mapping, which should be better and easier to maintain

Reply:
let me implement the later one

Reply:
updated